File: solutions/java/twoSumSorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
	int left = 0;
	int right = numbers.length - 1;

	while (left < right) {
	    int combined = numbers[left] + numbers[right];

	    if (combined == target) {
		return new int[]{left + 1, right + 1};
	    } else if (combined < target) {
		left++;
	    } else if (combined > target) {
		right--;
	    }
	}
	return new int[]{-1, -1};
    }
}

File: solutions/python/two_sum_sorted.py
class Solution:
    def two_sum(numbers: list[int], target:int) -> list[int]:
        left = 0
        right = len(numbers) - 1

        while left < right:
            combined = numbers[left] + numbers[right]
            if combined == target:
                return [left + 1, right + 1]
            elif combined < target:
                left += 1
            elif combined > target:
                right -= 1


    print(two_sum([2,7,11,15], 9))
    print(two_sum([2,3, 4], 6))
    print(two_sum([-1, 0], -1))
